### PR TITLE
Expand helm support with additional config

### PR DESCRIPTION
**Pull request by TiberiuGC**
Issue #, if available:

Description of changes:
- improve helm template for controller-deployment with additional config
- add metrics service

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: helm/templates/_helpers.tpl [hunk 1]
```diff
@@ -40,3 +40,14 @@ Create the name of the service account to use
...
+{{/* The mount path for the shared credentials file */}}
+{{- define "symphony.aws.credentials.secret_mount_path" -}}
+{{- "/var/run/secrets/aws" -}}
+{{- end -}}
+
+{{/* The path the shared credentials file is mounted */}}
+{{- define "symphony.aws.credentials.path" -}}
+{{ $secret_mount_path := include "symphony.aws.credentials.secret_mount_path" . }}
+{{- printf "%s/%s" $secret_mount_path .Values.aws.credentials.secretKey -}}
+{{- end -}}
```
Comment (a-hilaly) [inline helm/templates/_helpers.tpl]: Symphony doesn't need aws credentials (it doesn't communicate with AWS)

File: helm/templates/controller-deployment.yaml [hunk 2]
```diff
@@ -3,26 +3,128 @@ kind: Deployment
...
+              path: /healthz
+              port: 8079
+            initialDelaySeconds: 20
+            periodSeconds: 10
+          readinessProbe:
+            httpGet:
+              path: /readyz
+              port: 8079
+            initialDelaySeconds: 20
+            periodSeconds: 10
```
Comment (a-hilaly) [inline helm/templates/controller-deployment.yaml]: QQ: were you able to test the helm chart? not sure if healthz and readyz are enabled

File: helm/templates/controller-deployment.yaml [hunk 3]
```diff
@@ -3,26 +3,128 @@ kind: Deployment
...
+              mountPath: {{ include "symphony.aws.credentials.secret_mount_path" . }}
+              readOnly: true
+          {{- end }}
+          {{- if .Values.deployment.extraVolumeMounts -}}
+            {{ toYaml .Values.deployment.extraVolumeMounts | nindent 10 }}
+          {{- end }}
+          securityContext:
+            allowPrivilegeEscalation: false
+            privileged: false
+            # runAsNonRoot: true
```
Comment (a-hilaly) [inline helm/templates/controller-deployment.yaml]: Is there a reason we're commenting this one out?

File: helm/templates/controller-deployment.yaml [hunk 4]
```diff
@@ -3,26 +3,128 @@ kind: Deployment
...
+      {{- end }}
+      {{- if .Values.deployment.extraVolumes }}
+        {{ toYaml .Values.deployment.extraVolumes | indent 8}}
+      {{- end }}
       containers:
         - name: {{ .Chart.Name }}
           image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
           imagePullPolicy: {{ .Values.image.pullPolicy }}
+          ports:
+          - name: http
```
Comment (a-hilaly) [inline helm/templates/controller-deployment.yaml]:
```suggestion
          - name: metricsPort
```

File: helm/templates/controller-deployment.yaml (hunk 2 above)
Comment (TiberiuGC) [reply helm/templates/controller-deployment.yaml]:
```
kubectl get all
NAME                                            READY   STATUS             RESTARTS   AGE
pod/test-symphony-chart-8d5c7c64c-mwd9b         1/1     Running            0          11m
```

File: helm/templates/controller-deployment.yaml (hunk 3 above)
Comment (TiberiuGC) [reply helm/templates/controller-deployment.yaml]: Standard practice seems to be:
```
securityContext:
    runAsUser: 1000
    runAsNonRoot: true
```
Thanks for flagging this!

**Pull request merged**